hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/net/InnerNodeImpl.java

```diff
@@ -313,12 +314,12 @@ public Node getLeaf(int leafIndex) {
    *   means picking the 3th available node, which is n5.
```

Comment:
It is not related to the current change. But the example in the comments need further clarification. E.g., n5 is under /dc2 which is in excludedScope. Maybe you want to put n5 under /root/dc1/rack2? instead of /root/dc2/rack1?

Reply:
Right, currently no node can be selected given all the input parameters. I change the exclude scope from /dc2 to /dc2/rack2. Then the chosen node is /dc1/rack2/n5.